docs/design/security/README.adoc

```diff
@@ -0,0 +1,534 @@
+
+Hazelcast Persistence feature supports data encryption. It safeguards
+sensitive data stored on disk by encrypting it using industry-standard
+encryption algorithms. This feature helps protect data at rest from
+unauthorized access and enhances overall data security within the
+Hazelcast cluster.
+
+[source,yaml]
+----
+hazelcast:
+  map:
+    test-map:
+      data-persistence:
+        enabled: true
+  persistence:
+    enabled: true
+    base-dir: /mnt/persistence
+    backup-dir: /mnt/persistence-backup
+    encryption-at-rest:
+      enabled: true
+      algorithm: AES/CBC/PKCS5Padding
+      salt: NewYorkThymes
+      secure-store:
+        keystore:
+          path: /path/to/keystore.file
+          type: PKCS12
+          password: password
+----
+
+[[authentication-basics]]
+== Authentication
+
+Hazelcast supports authentication on Member protocol, Client protocol,
+and a subset of REST endpoints.
+
+Users can choose from a set of authentication mechanisms:
+
+* default (compares credentials against preconfigured member identity)
+* Simple - users and roles configured directly in the member
+configuration
+* TLS - available when mutual authentication is enabled
+* LDAP
+* Kerberos
+* JAAS
+
+[[member-to-member-authn]]
+=== Member-to-Member authentication
+
+In this sample configuration, we use the **default** authentication between
+cluster members. All members share the same identity in this case. We
+don't need to add the `authentication` section within the security
+`realm` configuration. The default authentication compares the incoming
+username and password to the member's `identity` configuration.
+
+[source,yaml]
+----
+hazelcast:
+  security:
+    enabled: true
+    realms:
+      - name: passwordRealm-members
+        identity:
+          username-password:
+             username: aUserNameOfYourChoice
+             password: PutAPasswordHere
+    # map the member protocol authentication to the defined realm
+    member-authentication:
+      realm: passwordRealm-members
+----
+
+[[client-to-member-authn]]
+=== Client-to-member authentication
+
+We can use a **simple** authentication method for clients. It allows
+fine-grained control over the role assignment. As this realm only verifies
+incoming client requests, there is no need to specify the member’s identity
+in the realm configuration.
+
+[source,yaml]
+----
+hazelcast:
+  security:
+    enabled: true
+    realms:
+      - name: simpleRealm-clients
+        authentication:
+          simple:
+            users:
+              - username: root
+                password: 7931verySecret
+                roles:
+                  - admin
+              - username: readonly
+                password: anotherSecret654
+                roles:
+                  - monitor
+    # map the client protocol authentication to the defined realms
+    client-authentication:
+      realm: simpleRealm-clients
+    client-permissions:
+      all:
+        principal: admin
+----
```

Comment:
Can we have a client side example as well? How would client config look like?

Reply:
:+1: I'll add a simple Java client config